### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by jmartinez95** [jmartinez95/MI-449-SS17-741-js-objects-T4W3KQ#1]
@chrisvfritz Can you take a look at this? It's [hosted here](https://mi-449-ss17-741-js-objects-t4w3kq-jmartinez95.surge.sh/) and meets the following criteria:

- [x] Your app builds on [these files](https://gist.github.com/chrisvfritz/3bd1b8d8e5b0d1ea2278ffbc561de047) (you can make changes to any of them)
- [x] When the `requestedJokeInput` has a value matching the key of an existing joke, the `jokeBox` HTML is updated [in this format](https://gist.github.com/chrisvfritz/4def3c246dfb931ecb57384a5639af99)
- [x] When the `requestedJokeInput` does _not_ match an existing joke, the `jokeBox` content is updated to `No matching joke found.`
- [x] When the `Remember this joke!` button is clicked, a new joke is saved (or overwritten) in the `jokes` object, using the data in the associated `input`/`textarea` elements
- [x] When the `Forget about it!` button is clicked, the joke with that key (if any) is deleted from the `jokes` object
- [x] Whenever the `jokes` object changes, the new version is saved in `localStorage` and used whenever the user visits the page again
- [x] It's impossible for an error to ever occur in the JavaScript console

Link is broken up top - correct version is

https://jmartinez95.github.io/MI-449-SS17-741-js-objects-T4W3KQ/

**Comment by chrisvfritz** [comment]
Looking good! 💥  The functionality is 100% perfect. 😄  The only areas for improvement are in the code itself:

### Repeated code

I notice a couple instances of this code exactly:

``` js
var stringifiedjokes = JSON.stringify(jokes)
window.localStorage.setItem('jokes',stringifiedjokes);
```

When we repeat the same code more than once, it's often a sign that it should be in its own function. In this case however, I notice another pattern. We also call `updatePage` afterwards every time:

``` js
updatePage();
```

That makes me think that we really might want an `updateJokes` function that saves the new jokes to localStorage and then calls `updatePage`, since we'll never want the page to be out of sync with the jokes.

### `stringifiedjokes` should be camelCase

You do a great job of sticking to camelCase for variable names most of the time - the only exception being `stringifiedjokes`.

### JS Standard style violations

I'm seeing quite a few style violations of JS Standard. Let me know if you're having any issues getting the plugin to work in Atom and I'll be happy to help. 🙂 

### Warning about `localStorage` with GitHub Pages

A quick note about using GitHub Pages instead of Surge with projects that use `localStorage`: since the data is scoped to the subdomain, _all your projects_ on GitHub pages will actually share the same data, which could allow them to mess with each other.

Did you encounter issues pushing with the Surge URL?

**Comment by jmartinez95** [comment]
@chrisvfritz Fixed them all except the surge command, doesn't seem to be working for me.

**Comment by chrisvfritz** [comment]
Looking better! 👍  I'm still seeing the JS Standard style violations though and I want to make sure we can get Surge working, for future projects - so reach out to me in class today when you settle in.

**Comment by jmartinez95** [comment]
Ah got surge working, I published it to the wrong domain...


https://mart1527-kaprwtcydf6wbt4w3kq.surge.sh/

as for the linter I literally have no idea what is still wrong?

![before and after](https://cloud.githubusercontent.com/assets/10256426/23186084/c3176a54-f853-11e6-8c95-3307d42d041e.png)

I have it working and I am getting a no issues, as you can see on the left when I remove and put the semi colon back.

@chrisvfritz

**Comment by chrisvfritz** [comment]
It looks like you have JSHint installed instead of JS Standard. See [this lesson page](https://msu.lansing.codes/courses/MI-449-SS17-741/lessons/js-collect-and-display-information/1) for instructions on installing the correct linter.

**Comment by jmartinez95** [comment]
@chrisvfritz I tried all of class last period with Katie and Erik to try and get the correct linter installed however it was not detecting issues. https://github.com/jmartinez95/MI-449-SS17-741-js-localstorage-wMUDce/issues/1

Take a look at what Erik said on this issue from the last project. There seems to be an issue with the linter that you are requesting that we use.

**Comment by chrisvfritz** [comment]
@jmartinez95 Yes, that day it was having some issues. The problems seem to have been resolved with an update later that night though, so if it's still not working for you, there's something else going on. Reach out to me on Slack (I'm `@chrisvfritz` there too) and we can do a screenshare to dig into what's going on.

**Comment by jmartinez95** [comment]
Got the linter to work must have just been that day or something. that should be good @chrisvfritz

**Comment by chrisvfritz** [comment]
Much better! 🎉 :shipit:
